### PR TITLE
Updating TSO500 DNA for new somatic nextflow pipeline

### PR DESCRIPTION
**Pull request by SophieS9**
Changes made to:

- Config file - different end point files for TSO500 DNA pipeline
- TSO500 parser - handling new files
- View Run Analysis html - now displays breakdown of FastQC checks as a "hovering box" above the FastQC result. This includes a new function in the models.  
- Unit tests and test data for TSO500 DNA.

Tested with outputs from 
/data/output/validations/somatic_enrichment_nextflow/210811_A00748_0131_AHFG2LDRXY_run3_new_gaps/
/data/output/validations/somatic_enrichment_nextflow/220502_A01771_0020_AH5TKWDRX2_run3_new_gaps
/data/output/validations/somatic_enrichment_nextflow/220608_A01771_0044_AH7VK5DRX2_run4_new_fastqc/
/data/output/validations/somatic_enrichment_nextflow/221114_A01771_0117_AHNLNYDRX2_run2_new_gaps/

Unit tests complete with no errors.

### CODE REVIEW

**Comment by SophieS9** [comment]
- Merged development branch, resolved conflicts and checked that everything is still working. All good :)
- ctDNA doesn't display the FastQC metrics when you hover over it, but all of the other runs do. 

Test data (one TSO500 and one ctDNA) on hard drive with sticker under AutoQC_SVD_Test/

**Comment by Istaisa** [comment]
Unit tests all pass.

TSO500 DNA runs from new pipeline upload as expected and display correct metrics.

Need to test:
- Full RNA/ctDNA run
- TSO500 DNA runs from old pipeline already in database still display correctly

File: config/config_local.yaml
Comment (Istaisa) [inline config/config_local.yaml:1]: Config changes needed for local config:

- Remove ] on line 51 
- Change the ctDNA local config to reflect the gen01 config

File: pipelines/TSO500_pipeline.py [hunk 1]
```diff
@@ -55,65 +56,74 @@ def run_is_valid(self):
...
-				found_file_list.append(file)
+				for x in outfile:
+					last_report = x.strip()
 
-		if len(list(found_file_list)) == len(self.run_expected_files):
+			if 'success' in last_report:
 
 				return True
 
+
```
[the remark this answers is not in the document]
Comment (Istaisa) [reply pipelines/TSO500_pipeline.py:87]: not necessary but might be useful if someone else picks this up in the future

**Comment by Istaisa** [comment]
update_database.py will need changing to account for the new TSO500 folder structure

line 504 should be something like `run_data_dir = Path(results_dir).joinpath(f"{run_analysis.run.run_id}/TSO500/")`

**Review by Istaisa — CHANGES_REQUESTED** [review]
- Unit tests all pass
- Manual testing of DNA, RNA and ctDNA runs all fine
- Hovering FastQC stats is great!
- Code logic makes sense

Summary of changes requested:
- Updates to local config
- Alteration to update_database.py for new folder structure
- Point about code comment is up to you

Unable to test that old TSO500 runs still display correctly due to quirk of changing git branch breaking local AutoQC but Sophie has tested - we'll check this when it goes live. I can't see anything in the code that makes me think this should be a problem though.

**Comment by SophieS9** [comment]
Code review changes all done as above. Changed the structure of the test data to reflect the changes.

**Comment by SophieS9** [comment]
Addition of minor bug fixes as code reviewed in https://github.com/AWGL/auto_qc/pull/157.

**Pull request merged**